test/e2e/installplan_e2e_test.go

```diff
@@ -2869,92 +2869,121 @@ var _ = Describe("Install Plan", func() {
 		// Should have removed every matching step
 		require.Equal(GinkgoT(), 0, len(expectedSteps), "Actual resource steps do not match expected: %#v", expectedSteps)
 
+		createdResources := map[string]struct{}{}
+
 		// the test from here out verifies created RBAC is removed after CSV deletion
 		createdClusterRoles, err := c.KubernetesInterface().RbacV1().ClusterRoles().List(context.Background(), metav1.ListOptions{LabelSelector: fmt.Sprintf("%v=%v", ownerutil.OwnerKey, stableCSVName)})
-		createdClusterRoleNames := map[string]struct{}{}
 		for _, role := range createdClusterRoles.Items {
-			createdClusterRoleNames[role.GetName()] = struct{}{}
+			createdResources["cr/"+role.GetName()] = struct{}{}
 			GinkgoT().Logf("Monitoring cluster role %v", role.GetName())
 		}
 
 		createdClusterRoleBindings, err := c.KubernetesInterface().RbacV1().ClusterRoleBindings().List(context.Background(), metav1.ListOptions{LabelSelector: fmt.Sprintf("%v=%v", ownerutil.OwnerKey, stableCSVName)})
-		createdClusterRoleBindingNames := map[string]struct{}{}
 		for _, binding := range createdClusterRoleBindings.Items {
-			createdClusterRoleBindingNames[binding.GetName()] = struct{}{}
+			createdResources["crb/"+binding.GetName()] = struct{}{}
 			GinkgoT().Logf("Monitoring cluster role binding %v", binding.GetName())
 		}
 
+		createdResources["sa/"+serviceAccountName] = struct{}{}
+		GinkgoT().Logf("Monitoring service account %v", serviceAccountName)
+
 		crWatcher, err := c.KubernetesInterface().RbacV1().ClusterRoles().Watch(context.Background(), metav1.ListOptions{LabelSelector: fmt.Sprintf("%v=%v", ownerutil.OwnerKey, stableCSVName)})
 		require.NoError(GinkgoT(), err)
 		crbWatcher, err := c.KubernetesInterface().RbacV1().ClusterRoleBindings().Watch(context.Background(), metav1.ListOptions{LabelSelector: fmt.Sprintf("%v=%v", ownerutil.OwnerKey, stableCSVName)})
 		require.NoError(GinkgoT(), err)
+		saWatcher, err := c.KubernetesInterface().CoreV1().ServiceAccounts(generatedNamespace.GetName()).Watch(context.Background(), metav1.ListOptions{})
```

Comment:
This would be a bigger change, but how would you feel about removing all these watches and replacing them with list calls inside of an eventually loop?

Reply:
Ambivalent. What are your thoughts here? Avoiding the watches (which should be more efficient) or the go routine?

Reply:
I'd rather poll. Otherwise you have to deal with watches getting broken for various reasons, and restarting them.

Reply:
Let me look at it. There are five resources spread over three types, it won't be pretty!

Reply:
So, it might actually not be that bad...